### PR TITLE
Update the Update and Bulk API's  update/upsert sections

### PR DESCRIPTION
**Pull request by kolchfa-aws**
Update the Update and Bulk API's  update/upsert sections

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _api-reference/document-apis/bulk.md [hunk 1]
```diff
@@ -82,7 +82,7 @@ All actions support the same metadata: `_index`, `_id`, and `_require_alias`. If
 
 - Create
 
-  Creates a document if it doesn't already exist and returns an error otherwise. The next line must include a JSON document.
+  Creates a document if it doesn't already exist and returns an error otherwise. The next line must include a JSON document:
```
Comment (natebower) [inline _api-reference/document-apis/bulk.md:85]: "but returns"?

File: _api-reference/document-apis/bulk.md [hunk 2]
```diff
@@ -91,15 +91,15 @@ All actions support the same metadata: `_index`, `_id`, and `_require_alias`. If
 
 - Delete
 
-  This action deletes a document if it exists. If the document doesn't exist, OpenSearch doesn't return an error, but instead returns `not_found` under `result`. Delete actions don't require documents on the next line.
+  This action deletes a document if it exists. If the document doesn't exist, OpenSearch doesn't return an error, but instead returns `not_found` under `result`. Delete actions don't require documents on the next line:
```
Comment (natebower) [inline _api-reference/document-apis/bulk.md:94]:
```suggestion
  This action deletes a document if it exists. If the document doesn't exist, OpenSearch doesn't return an error but instead returns `not_found` under `result`. Delete actions don't require documents on the next line:
```

Comment (natebower) [inline _api-reference/document-apis/bulk.md:94]: "documents to be on"?

File: _api-reference/document-apis/update-document.md [hunk 3]
```diff
@@ -11,7 +11,7 @@ redirect_from:
 **Introduced 1.0**
 {: .label .label-purple }
 
-If you need to update a document's fields in your index, you can use the update document API operation. You can do so by specifying the new data you want in your index or by including a script in your request body, which OpenSearch runs to update the document.
+If you need to update a document's fields in your index, you can use the update document API operation. You can do so by specifying the new data you want in your index or by including a script in your request body, which OpenSearch runs to update the document. By default, the update operation only updates a document that exists in the index. If a document does not exist, the API returns an error. To _upsert_ a document (update the document that exists or index a new one), use the [upsert](#upsert) operation.
```
Comment (natebower) [inline _api-reference/document-apis/update-document.md:14]: "you want to be in"?

File: _api-reference/document-apis/update-document.md [hunk 4]
```diff
@@ -65,7 +65,7 @@ wait_for_active_shards | String | The number of active shards that must be avail
 
 ## Request body
 
-Your request body must contain the information you want to update your document with. If you just want to replace certain fields in your document, your request body must include a `doc` object, which has the fields you want to update.
+Your request body must contain the information you want to update your document with. If you only want to replace certain fields in your document, your request body must include a `doc` object, which has the fields you want to update:
```
Comment (natebower) [inline _api-reference/document-apis/update-document.md:68]:
```suggestion
Your request body must contain the information with which you want to update your document. If you only want to replace certain fields in your document, your request body must include a `doc` object containing the fields that you want to update:
```

File: _api-reference/document-apis/update-document.md [hunk 5]
```diff
@@ -90,11 +90,14 @@ You can also use a script to tell OpenSearch how to update your document.
 }
 ```
 
-### Upsert
+## Upsert
 
-Upsert is an operation that conditionally either updates an existing document or inserts a new one based on information in the object. In the following example, the `upsert` operation updates the `last name` and adds the `first_name` field if a document already exists. If a document does not exist, a new one is indexed using content in the `upsert` object.
+Upsert is an operation that conditionally either updates an existing document or inserts a new one based on information in the object. 
+
+In the following example, the `upsert` operation updates the `last name` and adds the `first_name` field if a document already exists. If a document does not exist, a new one is indexed using content in the `upsert` object.
```
Comment (natebower) [inline _api-reference/document-apis/update-document.md:97]:
```suggestion
In the following example, the `upsert` operation updates the `last name` field and adds the `first_name` field if a document already exists. If a document does not exist, a new one is indexed using content in the `upsert` object.
```

File: _api-reference/document-apis/update-document.md [hunk 6]
```diff
@@ -106,9 +109,53 @@ Upsert is an operation that conditionally either updates an existing document or
...
+  "_id": "1",
+  "_score": 1,
+  "_source": {
+    "last_name": "Oliveira",
+    "age": "31"
+  }
+}
+```
+
+You can also add `doc_as_upsert` to the request and set it to `true` to use the information in `doc` for performing the upsert operation:
```
Comment (natebower) [inline _api-reference/document-apis/update-document.md:155]: "in the `doc` field"?

File: _api-reference/document-apis/bulk.md (hunk 1 above)
Comment (kolchfa-aws) [reply _api-reference/document-apis/bulk.md:85]: I think "and" works better here.

File: _api-reference/document-apis/update-document.md (hunk 5 above)
Comment (kolchfa-aws) [reply _api-reference/document-apis/update-document.md:97]: Reworded.

File: _api-reference/document-apis/update-document.md (hunk 6 above)
Comment (kolchfa-aws) [inline _api-reference/document-apis/update-document.md:155]:
```suggestion
You can also add `doc_as_upsert` to the request and set it to `true` to use the information in the `doc` field for performing the upsert operation:
```

**Review by vagimeli — APPROVED** [review]
LGTM

**Pull request merged**